src/main/java/duke/TaskList.java

```diff
@@ -129,4 +129,20 @@ Task getTaskAt(int taskIndex) {
     boolean isEmpty() {
         return taskArrayList.size() == 0;
     }
+
+    /**
+     * Retrieves a sublist of tasks whose descriptions contain the specified keyword
+     *
+     * @param keyword Keyword
+     * @return Sublist of tasks
+     */
+    public Task[] getSublistContainingKeyword(String keyword) {
+        ArrayList<Task> tasksContainingKeyword = new ArrayList<>();
+        for (Task task : this.taskArrayList) {
+            if (task.hasKeyword(keyword)) {
+                tasksContainingKeyword.add(task);
+            }
+        }
+        return tasksContainingKeyword.toArray(new Task[0]);
+    }
```

Comment:
Is there a particular reason you decided to convert ArrayList<Task> to Task[] before returning?